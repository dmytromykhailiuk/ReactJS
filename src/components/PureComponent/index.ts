export { default as PureComponent } from "./PureComponent";
