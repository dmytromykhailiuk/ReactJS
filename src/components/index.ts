export * from "./Component";
export * from "./PureComponent";
export * from "./FunctionalComponent";
