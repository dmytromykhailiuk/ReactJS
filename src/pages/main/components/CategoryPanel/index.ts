export { default as CategoryPanel } from './CategoryPanel';
