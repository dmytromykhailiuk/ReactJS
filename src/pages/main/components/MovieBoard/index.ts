export { default as MovieBoard } from "./MovieBoard";
export { default as MovieBoardContainer } from "./MovieBoardContainer";
