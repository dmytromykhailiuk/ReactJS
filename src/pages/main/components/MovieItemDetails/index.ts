export { default as MovieItemDetails } from "./MovieItemDetails";
