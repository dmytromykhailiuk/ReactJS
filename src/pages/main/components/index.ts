export * from "./Banner";
export * from "./MovieBoard";
export * from "./SearchPanel";
export * from "./CategoryPanel";
export * from "./SortPanel";
export * from "./MovieList";
export * from "./MovieItem";
export * from "./MovieMenu";
export * from "./MovieModal";
