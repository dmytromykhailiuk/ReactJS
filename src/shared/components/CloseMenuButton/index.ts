export { default as CloseMenuButton } from './CloseMenuButton';
