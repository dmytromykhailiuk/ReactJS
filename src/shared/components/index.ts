export * from "./Footer";
export * from "./Logo";
export * from "./Button";
export * from "./Input";
export * from "./ThreeDotsIcon";
export * from "./ErrorBoundary";
