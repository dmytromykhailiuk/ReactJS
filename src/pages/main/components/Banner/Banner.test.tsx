import { shallow } from "enzyme";
import React from "react";
import { Banner } from "./";
import toJson from 'enzyme-to-json';
import { Button } from "../../../../shared/components";

describe("Banner", () => {
  it("should match snepshot", () => {
    const wrapper = shallow(
      <Banner
        isMovieInOverviewMode
        onCreateMovie={() => {}}
        onLogoClicked={() => {}}
        onSearchIconClicked={() => {}}
      >
        Children
      </Banner>
    )
    expect(toJson(wrapper)).toMatchSnapshot();
  })

  it("should show Button if 'isMovieInOverviewMode' equal false ", () => {
    const wrapper = shallow(
      <Banner
        isMovieInOverviewMode={false}
        onCreateMovie={() => {}}
        onLogoClicked={() => {}}
        onSearchIconClicked={() => {}}
      >
        Children
      </Banner>
    )
    expect(toJson(wrapper.find(Button))).toBeTruthy();
  })
})
