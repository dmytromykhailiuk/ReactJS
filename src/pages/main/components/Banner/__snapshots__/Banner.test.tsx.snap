// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner should match first snepshot: first 1`] = `
<div
  className="banner"
>
  <div
    className="banner__image"
  />
  <div
    className="banner__background banner__background--blured"
  />
  <div
    className="banner__content"
  >
    <div
      className="banner__header"
    >
      <div
        className="banner__logo"
        onClick={[Function]}
      >
        <Logo />
      </div>
      <div
        className="banner__search-icon"
        onClick={[Function]}
      >
        <SearchIcon />
      </div>
    </div>
    Children
  </div>
</div>
`;
